tests/rbd_mirror/test_namespace_mirror_all_operations.py

```diff
@@ -0,0 +1,448 @@
+                    state_pattern="up+stopped",
+                )
+                # Verify image mirroring status on secondary cluster
+                wait_for_status(
+                    rbd=rbd_secondary,
+                    cluster_name=sec_config.get("cluster").name,
+                    imagespec=sec_image_spec,
+                    state_pattern="up+replaying",
+                )
+
+                snap_levels = image_config_val.get("snap_schedule_levels")
+                snap_intervals = image_config_val.get("snap_schedule_intervals")
+
+                if snap_levels and snap_intervals:
+                    level = snap_levels[0]
+                    interval = snap_intervals[0]
+
+                    # Handle case when level is 'namespace' but no namespace is defined (i.e., default ns)
+                    effective_level = (
+                        "image" if level == "namespace" and not namespace else level
+                    )
+
+                    out, err = add_snapshot_scheduling(
+                        rbd_primary,
+                        pool=pool,
+                        image=image,
+                        level=effective_level,
+                        interval=interval,
+                        namespace=namespace,
+                    )
+                    if err:
+                        raise Exception(
+                            "Adding snapshot schedule failed with error %s" % err
+                        )
+
+                    # Retain original level for remaining tests
+                    level = effective_level
+
+                    # if image_config_val.get(
+                    #     "snap_schedule_levels"
+                    # ) and image_config_val.get("snap_schedule_intervals"):
+                    #     for level, interval in zip(
+                    #         image_config_val["snap_schedule_levels"],
+                    #         image_config_val["snap_schedule_intervals"],
+                    #     ):
+                    #         snap_schedule_config = {
+                    #             "pool": pool,
+                    #             "image": image,
+                    #             "level": level,
+                    #             "interval": interval,
+                    #             "namespace": namespace,
+                    #         }
+                    #         # Adding snapshot schedules to the images in namespace
+                    #         out, err = add_snapshot_scheduling(
+                    #             rbd_primary, **snap_schedule_config
+                    #         )
+                    #         if err:
+                    #             raise Exception(
+                    #                 "Adding snapshot schedule failed with error " + err
+                    #             )
+                    # Verify snapshot schedules are effective on the namespaces
+                    verify_namespace_snapshot_schedule(
```

Comment:
@aarsharm use this method https://github.com/red-hat-storage/cephci/blob/main/ceph/rbd/workflows/snap_scheduling.py#L116
for random logic that method won't work.